internal/pkg/kernel/kernel.go

```diff
@@ -184,6 +185,23 @@ func (c *cmdline) Append(k string, v string) {
 	insert(&c.Parameters, k, v)
 }
 
+// AppendAll appends a set of kernel parameters.
+func (c *cmdline) AppendAll(args []string) error {
+	for _, arg := range args {
+		parts := strings.Split(arg, "=")
+		switch len(parts) {
+		case 1:
+			c.Append(parts[0], "")
+		case 2:
+			c.Append(parts[0], parts[1])
+		default:
+			return errors.New("expected kernel argument in key=value format")
```

Comment:
I believe argument might contain `=`, so it's more correct to do `parts := strings.SplitN(arg, "=", 2)` 

and use "" if parts[1] is missing

Reply:
If I understand you, I think we don't need to do this because `c.Append` adds the `=`.

Reply:
In order to support one key with multiple values, we use the follow type:

```
type Parameter struct {
	key    Key
	values []string
}
```
So the `=` doesn't exist until the `String()` method is called.